api/envoy/config/route/v3/route_components.proto

```diff
@@ -97,8 +96,7 @@ message VirtualHost {
   // [#next-major-version: This should be included in a oneof with routes wrapped in a message.]
```

Comment:
Given we don't expect a next major version should we clean this up too?

Reply:
I think this can be removed, but should be replaced with oneof-promotion tag. Not because it will ever change, but because it helps detecting all those fields that should be oneofs.

Somewhat related, I'd like to gradually change the route data-structure that we are using at the moment, so instead of including all the fields of a oneof in the same level in the data-structure, refactor them into different implementations of some interface (using the Strategy pattern). This should save some memory, and hopefully make the code a bit cleaner.

Reply:
yep.  `oneof_promotion` is standard approach for this case. It probably should be done in this way when `matcher` was introduced. 

Done.